### ISSUE TITLE
Suggestion: Replace chagetype<T> to "as unknown as T"

### ISSUE THREAD

**Issue by MaxGraey** [AssemblyScript/assemblyscript#1497]
What about use more idimatic:
```ts
function some_unsafe_cast(ptr: usize): T {
  return ptr as unknown as T;
  // or 
  // return <T><unknown>ptr;
}
```
instead:
```ts
function some_unsafe_cast(ptr: usize): T {
  return changetype<T>(ptr);
}
```
Thoughts?

**Comment by dcodeIO** [comment]
Hmm, `changetype` is an uncommon unsafe operation, and as it is feels like it, while `ptr as unknown` is surprising to fail, and `ptr as unknown as T` looks strange. Not sure.

**Comment by MaxGraey** [comment]
`ptr as unknown as T` in term of TS also unsafe and means "unsound cast". It's like we clear type (`as unknown` part) and assign new type (`as T` part)

**Comment by dcodeIO** [comment]
Hmm, perhaps as a hardcoded option for those preferring it. Wdyt?

**Comment by MaxGraey** [comment]
Yeah, it could be hardcoded as MVP. But will be great have full support of `unknown` type and casting rules for it
